### DISCUSSION TITLE
How to combine the 4 of 5 responses as a $ref?

### DISCUSSION THREAD

**Discussion by Yao544303** [OAI/OpenAPI-Specification#4231] [Q&A] [answered]
Hi. I am draft the API specification. Here is my question?
For each API, there are 5 responses, such as 
```
...
"responses":{
   "200":{
       // the 200 responses if different for each API
   },
   "400":{
      "$ref": "#/components/schemas/Error400"
   },
   "401":{
      "$ref": "#/components/schemas/Error401"
   },
   "404":{
      "$ref": "#/components/schemas/Error404"
   },
   "429":{
      "$ref": "#/components/schemas/Error429"
   }
}
...
```

Since the 200 is different for each API with different response schema, but the 400-429 is same. May I combine 400-429 into one $ref?
Such as 
```
"responses":{
   "200":{
       // the 200 responses if different for each API
   },
  "4xx":{
      "$ref":"xxx"
  }
}
...
```

**Comment by lornajane** [comment, marked as answer]
In OpenAPI, we don't have this feature, but you could use an Overlay to add the group as a separate step. There's a similar-ish example [in the Overlays docs](https://learn.openapis.org/overlay/example-add-filter-parameters.html) but please let us know if you need more context on what this is and how to do it?

**Reply by Yao544303** [reply]
This is great! Thank you.

**Comment by handrews** [comment]
See also:
* #2936

**Reply by Montiwa11** [reply]
Ok thank you

**Comment by handrews** [comment]
Closing as answered.
